lighty-modules/lighty-aaa-aggregator/lighty-aaa/src/main/java/io/lighty/aaa/AAALighty.java

```diff
@@ -10,60 +10,185 @@
+                = new ShiroWebEnvironmentLoaderListener(shiroConfiguration, dataBroker, certificateManager,
+                authenticationService, tokenAuthenticators, tokenStore, defaultPasswordHashService);
+        server.addCommonEventListener(shiroWebEnvironmentLoaderListener);
 
-        AAALightyShiroProvider aaaLightyShiroProvider;
+        final CustomFilterAdapterConfigurationImpl customFilterAdapterConfig =
+                new CustomFilterAdapterConfigurationImpl();
+        customFilterAdapterConfig.update(new HashMap<>());
+        final FilterHolder customFilterAdapter = new FilterHolder(new CustomFilterAdapter(customFilterAdapterConfig));
+        server.addCommonFilter(customFilterAdapter, "/*");
 
-        void setAaaLightyShiroProvider(final AAALightyShiroProvider aaaLightyShiroProvider) {
-            this.aaaLightyShiroProvider = aaaLightyShiroProvider;
+        final FilterHolder shiroFilter = new FilterHolder(new AAAShiroFilter());
+        server.addCommonFilter(shiroFilter, "/*");
+
+        final FilterHolder crossOriginFilter = new FilterHolder(new CrossOriginFilter());
+        crossOriginFilter.setInitParameter("allowedMethods", "GET,POST,OPTIONS,DELETE,PUT,HEAD");
+        crossOriginFilter.setInitParameter("allowedHeaders",
+                "origin, content-type, accept, authorization, Authorization");
+        server.addCommonFilter(crossOriginFilter, "/*");
+    }
+
+    @Override
+    protected boolean initProcedure() throws InterruptedException {
+        if (credentialAuth == null) {
+            final IdmLightProxy idmLightProxy = new IdmLightProxy(iidmStore, defaultPasswordHashService);
+            this.credentialAuth = idmLightProxy;
+            this.claimCache = idmLightProxy;
         }
+        this.tokenAuthenticators = buildTokenAuthenticators(this.credentialAuth);
+        try {
+            final StoreBuilder storeBuilder = new StoreBuilder(iidmStore);
+            final String domain = storeBuilder.initDomainAndRolesWithoutUsers(IIDMStore.DEFAULT_DOMAIN);
+            if (domain != null) {
+                // If is not exist. If domain already exist on path, will be used instead
+                storeBuilder.createUser(domain, aaaConfiguration.getUsername(), aaaConfiguration.getPassword(), true);
+            }
+        } catch (final IDMStoreException e) {
+            LOG.error("Failed to initialize data in store", e);
+            return false;
+        }
+        final LocalHttpServer httpService = new LocalHttpServer(server);
+        registerServletContexts(httpService, aaaConfiguration.getMoonEndpointPath());
+
+        initAAAonServer();
+        return true;
+    }
 
-        AAALightyShiroProvider getAaaLightyShiroProvider() {
-            return this.aaaLightyShiroProvider;
+    @Override
+    @SuppressWarnings("IllegalCatch")
+    protected boolean stopProcedure() {
+        final AtomicBoolean thrownException = new AtomicBoolean(true);
```

Comment:
Please rename variable `thrownException` to more proper name. Because if this variable is equals to `true` that means non exception was thrown.

Reply:
naming changed to `handlersStoppedSuccessfully`